Compilation error on Centos7

### Describe the current behavior

When compiling project with gcc 4.8.5 (on Centos 7) there are some compilation issues.  
This happens for 1.6.0 tag and above.

### Describe the expected behavior

The compilation should pass

### Describe the steps

Compile the project on tag 1.6.0

### Environment

- OS : centos 7
- gcc version 4.8.5
- cmake version 3.26.4

### Relevant Log Output

<pre>/home/barresthi/local/convergence_docker/powsybl-iidm4cpp/powsybl-iidm4cpp/extensions/iidm/src/BranchObservabilityAdder.cpp: In member function ‘virtual std::unique_ptr&lt;powsybl::iidm::Extension&gt; powsybl::iidm::extensions::iidm::BranchObservabilityAdder::createExtension(powsybl::iidm::Extendable&amp;) const’:
/home/barresthi/local/convergence_docker/powsybl-iidm4cpp/powsybl-iidm4cpp/extensions/iidm/src/BranchObservabilityAdder.cpp:42:16: error: could not convert ‘extension’ from ‘std::unique_ptr&lt;powsybl::iidm::extensions::iidm::BranchObservability&gt;’ to ‘std::unique_ptr&lt;powsybl::iidm::Extension&gt;’
         return extension;
                ^
/home/barresthi/local/convergence_docker/powsybl-iidm4cpp/powsybl-iidm4cpp/extensions/iidm/src/InjectionObservabilityAdder.cpp: In member function ‘virtual std::unique_ptr&lt;powsybl::iidm::Extension&gt; powsybl::iidm::extensions::iidm::InjectionObservabilityAdder::createExtension(powsybl::iidm::Extendable&amp;) const’:
/home/barresthi/local/convergence_docker/powsybl-iidm4cpp/powsybl-iidm4cpp/extensions/iidm/src/InjectionObservabilityAdder.cpp:40:16: error: could not convert ‘extension’ from ‘std::unique_ptr&lt;powsybl::iidm::extensions::iidm::InjectionObservability&gt;’ to ‘std::unique_ptr&lt;powsybl::iidm::Extension&gt;’
         return extension;
                ^
</pre>

### Extra Information

in files extensions/iidm/src/BranchObservabilityAdder.cpp and extensions/iidm/src/InjectionObservabilityAdder.cpp, if I use :  
`return std::move(extension);`  
instead of :  
`return extension;` 
it works well.

--- a/extensions/iidm/src/BranchObservabilityAdder.cpp
+++ b/extensions/iidm/src/BranchObservabilityAdder.cpp
@@ -39,7 +39,8 @@ std::unique_ptr<Extension> BranchObservabilityAdder::createExtension(Extendable&
         if (!std::isnan(m_standardDeviationQ2)) {
             extension->setQualityQ2(m_standardDeviationQ2, m_redundantQ2);
         }
-        return extension;
+        std::unique_ptr<Extension> baseExtension = std::move(extension);
+        return baseExtension;
     }
     throw AssertionError(stdcxx::format("Unexpected extendable type: %1% (%2% expected)", stdcxx::demangle(extendable), stdcxx::demangle<Branch>()));
 }

--- a/extensions/iidm/src/InjectionObservabilityAdder.cpp
+++ b/extensions/iidm/src/InjectionObservabilityAdder.cpp
@@ -37,7 +37,8 @@ std::unique_ptr<Extension> InjectionObservabilityAdder::createExtension(Extendab
         if (!std::isnan(m_standardDeviationV)) {
             extension->setQualityV(m_standardDeviationV, m_redundantV);
         }
-        return extension;
+        std::unique_ptr<Extension> baseExtension = std::move(extension);
+        return baseExtension;
     }
     throw AssertionError(stdcxx::format("Unexpected extendable type: %1% (%2% expected)", stdcxx::demangle(extendable), stdcxx::demangle<Injection>()));
 }

